core/vm/evm.go

```diff
@@ -234,9 +238,7 @@ func (self *EVM) Main(trx *Transaction) (ret ExecutionResult) {
 	}
 	if err != nil {
 		if err_str := util.ErrorString(err.Error()); err_str == ErrInsufficientBalanceForTransfer {
-			// set it back to original as we have ConsensusErr here
-			gas_left = gas_cap
-			ret.ConsensusErr = err_str
+			return consensusErr(gas_cap, err_str)
```

Comment:
are you sure about this change ? cannot we have some gas_left that should be returned to the acc balance ? As you are returning here, that will never happen

Reply:
We agreed that we are spending gas for all failed transactions. So it is a purpose of this whole PR to spend gas in places like this :)